### PR TITLE
MSG_LOG

### PR DESCRIPTION
**Pull request by mfine**
Introduce MSG_LOG, per #167. Note the values, which match the current logging levels in [libswiftnav] (https://github.com/swift-nav/libswiftnav/blob/master/include/libswiftnav/logging.h#L37-52). Also note the maskable id - is this where we want log messages, or do we want to give them their own group (0x0500?).

/cc @fnoble @denniszollo @mookerji @gsmcmullin

### CODE REVIEW

**Comment by mookerji** [comment]
lgtm!

**Comment by fnoble** [comment]
awesome, lgtm!

**Comment by mookerji** [comment]
To clarify, I don't have a preference much for the message masking for the non-obs type stuff. It's not obvious to me that they need to be treated any differently than the other things that don't go over the radio.

**Comment by fnoble** [comment]
oh, didn't consider the masking - let me have a think about that first. I think 0x01xx was already taken for a particular group.

**Comment by mfine** [comment]
Anyone else have preferences on the logging message mask stuff? @henryhallam? @gsmcmullin? @cbeighley? Reproducing Dennis' concern here:

*So I think that the MSG_LOG should have it's own group in the messages (i.e away from MSG_UART_STATE, MSG_THREAD_STATE). I sort of see that the product will only use this message if something pretty bad is happening for which we might need user awareness and user interaction.*

*I'm thinking a message ID of 0x110, 0x510 might work well, that way it comes with basic system event info (i.e. the external event trigger), the old logging group, and a new msg_log only bit.*

*This way you could get the warnings and errors on the device without all the thread state overhead.*

**Comment by fnoble** [comment]
So far we have the following groups, roughly speaking:
```
0000 00GG xGxx xxxx
       ||  | over the radio (0x40)
       ||--- time / event related messages
       |---- position / velocity related messages
```
G - group indicator bit
0 - unallocated group bit
x - sub-group indicator bit

We should probably establish more of a policy around this soon.

**Comment by fnoble** [comment]
With that in mind though, as seeing as this is being stabilized, I'd propose using 0x0401 and allocating the 11th bit of the ID for such informational kinds of messages.

File: spec/yaml/swiftnav/sbp/logging.yaml [hunk 1]
```diff
@@ -12,20 +12,33 @@ package: swiftnav.sbp.logging
 description: |
   Logging and debugging messages from the device. These are in the
   implementation-defined range (0x0000-0x00FF).
-stable: False
+stable: True
```
Comment (fnoble) [inline spec/yaml/swiftnav/sbp/logging.yaml]: Should stable be specified for the whole package? Will this make the deprecated message stable?

Comment (mfine) [inline spec/yaml/swiftnav/sbp/logging.yaml]: The stability will not apply to deprecated messages. I need to update that here and in a couple of other places. But, yes, `MSG_TWEET` will be stable.

Comment (fnoble) [inline spec/yaml/swiftnav/sbp/logging.yaml]: Is stability only indicated on a per package basis?

**Comment by mfine** [comment]
Yeah, it's only been indicated on a per package basis so far. I added per message type for public, we could do that for stability as well.

**Comment by henryhallam** [comment]
Stability should be per message.
0x0401 sounds good to me; we should move to a whitelist system rather than the message mask in the not too distant future.

**Comment by denniszollo** [comment]
I like 0x410, but think there might be more accidental implicit groups than the one's Fergus outlined.  Last time I looked it looked like the first few nibbles could be used and have some patterns.

**Comment by mfine** [comment]
0x401 or 0x410? Going to change and bring in

**Comment by mfine** [comment]
retest this please

**Pull request merged**